input_writer.go

```diff
@@ -0,0 +1,30 @@
+package logrus
+
+import (
+	"bufio"
+	"io"
+	"runtime"
+)
+
+func (logger *Logger) InputWriter() (*io.PipeWriter) {
+	inputReader, inputWriter := io.Pipe()
+
+	go logger.inputWriterScanner(inputReader)
+	runtime.SetFinalizer(inputWriter, inputWriterFinalizer)
+
+	return inputWriter
+}
+
+func (logger *Logger) inputWriterScanner(inputReader *io.PipeReader) {
+	scanner := bufio.NewScanner(inputReader)
+	for scanner.Scan() {
+		logger.Print(scanner.Text())
+	}
+	if err := scanner.Err(); err != nil {
+		logger.Fatal(err)
```

Comment:
when the `inputWriter` get closed, I think this end will get an `err == io.EOF`, so this would `os.Exit` the process as soon as the external writer stops writing.